fem/bilininteg_hcurl.cpp

```diff
@@ -366,6 +366,130 @@ void PAHcurlMassAssembleDiagonal3D(const int D1D,
    }); // end of element loop
 }
 
+void SmemPAHcurlMassAssembleDiagonal3D(const int D1D,
+                                       const int Q1D,
+                                       const int NE,
+                                       const Array<double> &_Bo,
```

Comment:
We encountered issues with variables starting with `_` and capital letters: a pass should be done to remove them.
We can discuss if the group has a preference.

Reply:
Done.